### PR TITLE
podman 4.4.0

### PR DESCRIPTION
**Pull request by xxyzz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
- Build with the latest go
- Use GNU Make to fix "invalid `override' directive" error
- Build from git tag to fix "fatal: not a git repository" error
- Fix build doc error by using just one job with `-j1` argument

### CODE REVIEW

**Comment by p-linnane** [comment]
Duplicate of https://github.com/Homebrew/homebrew-core/pull/122122.

**Comment by xxyzz** [comment]
I know there is a pr before but that pr hasn't changed a few days so I want to fix the build error in a new pr. Now at least the build is successful for macOS.

**Comment by xxyzz** [comment]
Please reopen and review this pull request, I have pushed two commits to fix the build and test error on Linux. @p-linnane

**Comment by iMichka** [comment]
Is the `-j1` to fix the docs build? We should use the patch from #122122 instead of using this workaround.

**Comment by xxyzz** [comment]
> Is the `-j1` to fix the docs build? We should use the patch from #122122 instead of using this workaround.

That patch won't fix the doc build as the #122122 log shows. podman's Makefile would fail when building with multiple jobs. Please see https://github.com/containers/podman/pull/17367

**Comment by iMichka** [comment]
Can we use your patch?

**Comment by xxyzz** [comment]
But that would need to apply two patches and my patch is not merged yet. Besides there is no benefit to build doc with multiple processes as the podman developer said in the first patch: https://github.com/containers/podman/commit/bd8937c5049771a60c40206fadda44316f1514ad

**Comment by iMichka** [comment]
We can ship patches that are not merged yet. We can also ship multiple patches, that's not an issues.

I understand that building docs with multiple processes is probably useless in this case and I can live with that. Upstream could also put the j1 in their build setup instead of us having to carry specific flags around.

We prefer upstream decides / manages their own compiler flags, because if they change their mind, the change will be propagated to us on the next update. Else we will keep useless / wrong flags forever on our side.

If it's too big of a hassle, we can still consider merging as-is, I'm not going to block this pull request on that.

**Comment by xxyzz** [comment]
podman's doc doesn't has any flag: https://podman.io/getting-started/installation#build-and-run-dependencies

So they are probably using the default make setting which is executing build tasks one at a time. The `-j1` flag actually suppresses the `HOMEBREW_MAKE_JOBS` flag to restore the default make behavior. I'll add a comment of removing this flag at next release.

File: Formula/podman.rb [hunk 1]
```diff
@@ -62,21 +63,22 @@ class Podman < Formula
...
-      sha256 "6dd1ce4346ed5c57bbd990140e02e69c036919032582b937d2ad7835329d3bc3"
+      url "https://github.com/containers/aardvark-dns/archive/refs/tags/v1.5.0.tar.gz"
+      sha256 "b7e7ca1b94c1a62c8800f49befb803ec37cc5caf7656352537343a8fb654e4a6"
     end
   end
 
   def install
     if OS.mac?
       ENV["CGO_ENABLED"] = "1"
+      ENV.prepend_path "PATH", Formula["make"].opt_libexec/"gnubin"
```
Comment (dawidd6) [inline Formula/podman.rb:81]: Why is `make` defined as a dependency on Linux above and it is prepended to PATH only on macOS? Also it should not be necessary to install brewed make on Linux, we require it to be installed system-wide, macOS on the other hand - sure.

Comment (xxyzz) [reply Formula/podman.rb:81]: I was meant to add `make` to the `on_macos` block but added it to `on_linux` by mistake. Thanks!

**Review by dawidd6 — APPROVED** [review]

**Review by iMichka — APPROVED** [review]
Cool, thanks!

**Pull request closed without merging**